### DISCUSSION TITLE
question: Customing request only work for async?

### DISCUSSION THREAD

**Discussion by 1xuan** [fastapi/fastapi#7036] [Questions]
Official doc say the feature [Custom Request and APIRoute class](https://fastapi.tiangolo.com/advanced/custom-request-and-route/)

And I've tried out the demo code. I found custom route class, custom request function and the api function run in same thread.(and api function is without async). So all threads will be blocked when one thread blocked. I wanna know whether or not there is a way I can use it without async?

- Python 3.8
- fastapi 0.61.1

**Comment by 1xuan** [comment]
It's a wrong question! My bad!

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
